### PR TITLE
refine write interface; remove deltaType property

### PR DESCRIPTION
**Pull request by jixuan1989**
Now the write API is faaaaaaaaaaar concise than before.
Example:
```java
		TsFileWriter tsFileWriter=new TsFileWriter(new File("test.ts"));
		tsFileWriter.addMeasurement(new MeasurementDescriptor("cpu_utility", TSDataType.FLOAT, TSEncoding.TS_2DIFF));
		tsFileWriter.addMeasurement(new MeasurementDescriptor("memory_utility", TSDataType.FLOAT, TSEncoding.TS_2DIFF));		
		TSRecord tsRecord=new TSRecord(1000, "hxd");
		DataPoint dPoint1=new FloatDataPoint("cpu_utility", 90.0f);
		DataPoint dPoint2=new FloatDataPoint("memory_utility", 80.0f);
		tsRecord.addTuple(dPoint1);
		tsRecord.addTuple(dPoint2);
		tsFileWriter.write(tsRecord);
		tsFileWriter.close();
```

### CODE REVIEW

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/TsFileWriter.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+/**
+ * TsFileWriter is the entrance for writing processing. It
+ * receives a record in type of {@code T} and send it to responding row group
+ * write. It checks memory size for all writing processing along its strategy
+ * and flush data stored in memory to OutputStream. At the end of writing, user
+ * should call {@code close()} method to flush the last data outside and close
+ * the normal outputStream and error outputStream.
+ *
+ * @param <T> - record type
```
Comment (Beyyes) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/TsFileWriter.java]: This ```<T>``` is wrong.

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java [hunk 2]
```diff
@@ -37,25 +39,36 @@
...
+    protected List<RowGroupMetaData> rowGroupMetaDatas = new ArrayList<>();
+    private RowGroupMetaData currentRowGroupMetaData;
+    private TimeSeriesChunkMetaData currentChunkMetaData;
 
+    
     /**
-     * This is just used to restore one TSFile from List of RowGroupMetaData and the offset
+     * for writing a new tsfile
+     *
+     * @param schema schema containing measurement information
```
Comment (Beyyes) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java]: The param ```schema``` and ```output```` is wrong...

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java [hunk 3]
```diff
@@ -37,25 +39,36 @@
...
+     */
+    public TSFileIOWriter(File file) throws IOException {
+        this.out = new TsRandomAccessFileWriter(file);
+        startFile();
+    }
+    
+    /**
+     * for writing a new tsfile.
      *
      * @param schema schema containing measurement information
```
Comment (Beyyes) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java]: The param ```schema```  is wrong. Some functions parameters below is also wrong.

File: src/main/java/cn/edu/tsinghua/tsfile/file/metadata/TimeSeriesMetadata.java [hunk 4]
```diff
@@ -46,17 +41,16 @@
...
         this.type = dataType;
-        this.deltaObjectType = deltaObjectType;
     }
 
     @Override
     public TimeSeries convertToThrift() {
         try {
             TimeSeries timeSeriesInThrift = new TimeSeries(measurementUID,
-                    type == null ? null : DataType.valueOf(type.toString()), deltaObjectType);
+                    type == null ? null : DataType.valueOf(type.toString()), "");//FIXME remove deltaType from TimeSeries.java
```
[the remark this answers is not in the document]
Comment (jixuan1989) [reply src/main/java/cn/edu/tsinghua/tsfile/file/metadata/TimeSeriesMetadata.java]: FIXME is a label in Eclipse for reminding you  fix this problem..
We have to modify thrift file to remove deltaType.

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/TsFileWriter.java [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+	public void write(TSRecord record) throws IOException, WriteProcessException {
+		if (checkIsDeltaExist(record)) {
+			groupWriters.get(record.deltaObjectId).write(record.time, record.dataPointList);
+			++recordCount;
+			checkMemorySize();
+		}
+	}
+
+	/**
+	 * @see cn.edu.tsinghua.tsfile.timeseries.write.seriesIRowGroupWriter#getDataInMemory(String)
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/TsFileWriter.java]: javadoc引用错误

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/desc/MeasurementDescriptor.java [hunk 6]
```diff
@@ -38,14 +39,14 @@
...
-        this.type = type;
-        this.measurementId = measurementId;
-        this.encoding = encoding;
-        this.conf = TSFileDescriptor.getInstance().getConfig();
+    	this(measurementId, type, encoding, Collections.emptyMap());
     }
 
     public MeasurementDescriptor(String measurementId, TSDataType type, TSEncoding encoding, Map<String, String> props) {
-        this(measurementId, type, encoding);
+    	 	 this.type = type;
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/desc/MeasurementDescriptor.java]: 缩进

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java (hunk 2 above)
Comment (qiaojialin) [reply src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java]: MAGIC_STRING 版本没更新，或者去掉版本号

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java [hunk 7]
```diff
@@ -37,25 +39,35 @@
...
-     * This is just used to restore one TSFile from List of RowGroupMetaData and the offset
+     * for writing a new tsfile
      *
      * @param schema schema containing measurement information
      * @param output be used to output written data
      * @throws IOException if I/O error occurs
      */
-    public TSFileIOWriter(FileSchema schema, TSRandomAccessFileWriter output) throws IOException {
-        this.schema = schema;
+    public TSFileIOWriter(File file) throws IOException {
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java]: javadoc没改

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java [hunk 8]
```diff
@@ -37,25 +39,35 @@
...
+     */
+    public TSFileIOWriter(ITsRandomAccessFileWriter output) throws IOException {
         this.out = output;
         startFile();
     }
-
+   
     /**
      * This is just used to restore one TSFile from List of RowGroupMetaData and the offset
      *
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java]: javadoc

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java [hunk 9]
```diff
@@ -96,12 +107,11 @@ private void startFile() throws IOException {
      * @param deltaObjectType - delta type of this row group
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TSFileIOWriter.java]: 参数不存在了，去掉javadoc

File: src/test/java/cn/edu/tsinghua/tsfile/file/metadata/utils/Utils.java [hunk 10]
```diff
@@ -65,11 +65,6 @@ public static void isTimeSeriesEqual(TimeSeriesMetadata timeSeriesInTSF,
       assertTrue(
           timeSeriesInTSF.getMeasurementUID().equals(timeSeriesInThrift.getMeasurement_uid()));
     }
-    if (Utils.isTwoObjectsNotNULL(timeSeriesInTSF.getDeltaObjectType(),
-        timeSeriesInThrift.getDelta_object_type(), "device type")) {
-      assertTrue(
-          timeSeriesInTSF.getDeltaObjectType().equals(timeSeriesInThrift.getDelta_object_type()));
-    }
     assertTrue(timeSeriesInTSF.getTypeLength() == timeSeriesInThrift.getType_length());
```
Comment (qiaojialin) [inline src/test/java/cn/edu/tsinghua/tsfile/file/metadata/utils/Utils.java]: String.equals()

File: src/test/java/cn/edu/tsinghua/tsfile/timeseries/demo/WriteATsfile.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+import cn.edu.tsinghua.tsfile.file.metadata.enums.TSDataType;
+import cn.edu.tsinghua.tsfile.file.metadata.enums.TSEncoding;
+import cn.edu.tsinghua.tsfile.timeseries.write.TsFileWriter;
+import cn.edu.tsinghua.tsfile.timeseries.write.desc.MeasurementDescriptor;
+import cn.edu.tsinghua.tsfile.timeseries.write.exception.WriteProcessException;
+import cn.edu.tsinghua.tsfile.timeseries.write.record.DataPoint;
+import cn.edu.tsinghua.tsfile.timeseries.write.record.TSRecord;
+import cn.edu.tsinghua.tsfile.timeseries.write.record.datapoint.FloatDataPoint;
+
+public class WriteATsfile {
```
Comment (qiaojialin) [inline src/test/java/cn/edu/tsinghua/tsfile/timeseries/demo/WriteATsfile.java]: WriteTsFile

File: src/test/java/cn/edu/tsinghua/tsfile/timeseries/generator/GenerateBigTSFile.java [hunk 12]
```diff
@@ -1,37 +1,32 @@
...
-import java.io.File;
-import java.io.IOException;
-import java.util.HashSet;
-import java.util.Set;
+import cn.edu.tsinghua.tsfile.timeseries.write.schema.FileSchema;
 
 public class GenerateBigTSFile {
     private static final Logger LOG = LoggerFactory.getLogger(GenerateBigTSFile.class);
-    private static InternalRecordWriter<TSRecord> innerWriter;
+    private static TsFileWriter innerWriter;
```
Comment (qiaojialin) [inline src/test/java/cn/edu/tsinghua/tsfile/timeseries/generator/GenerateBigTSFile.java]: 把变量名顺便改了吧

File: src/test/java/cn/edu/tsinghua/tsfile/file/metadata/utils/Utils.java (hunk 10 above)
Comment (jixuan1989) [reply src/test/java/cn/edu/tsinghua/tsfile/file/metadata/utils/Utils.java]: getTypeLength() return a int

**Comment by jixuan1989** [comment]
@qiaojialin  done

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/TsFileWriter.java [hunk 13]
```diff
@@ -0,0 +1,310 @@
...
+  /**
+   * Confirm whether the record is legal. If legal, add it into this RecordWriter.
+   *
+   * @param record
+   *          - a record responding a line
+   * @return - whether the record has been added into RecordWriter legally
+   * @throws IOException
+   *           exception in IO
+   */
+  protected boolean checkIsDeltaExist(TSRecord record) throws IOException, WriteProcessException {
```
Comment (kr11) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/TsFileWriter.java]: IOException is unnecessary

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+   * start a {@linkplain RowGroupMetaData RowGroupMetaData}.
+   *
+   * @param recordCount
+   *          - the record count of this time series input in this stage
+   * @param deltaObjectId
+   *          - delta object id
+   * @throws IOException
+   *           if I/O error occurs
+   */
+  public void startRowGroup(long recordCount, String deltaObjectId) throws IOException {
```
Comment (kr11) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java]: The IOException is unnecessary.

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+
+    VInTimeSeriesChunkMetaData v = new VInTimeSeriesChunkMetaData(tsDataType);
+    TSDigest tsDigest = new TSDigest(ByteBuffer.wrap(max, 0, max.length),
+        ByteBuffer.wrap(min, 0, min.length));
+    v.setDigest(tsDigest);
+    descriptor.setDataValues(v);
+    currentChunkMetaData.setVInTimeSeriesChunkMetaData(v);
+  }
+
+  public void endSeries(long size, long totalValueCount) throws IOException {
```
Comment (kr11) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java]: The IOException is unnecessary.

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+    LOG.debug("end series:{},totalvalue:{}", currentChunkMetaData, totalValueCount);
+    currentChunkMetaData.setTotalByteSize(size);
+    currentChunkMetaData.setNumRows(totalValueCount);
+    currentRowGroupMetaData.addTimeSeriesChunkMetaData(currentChunkMetaData);
+    currentChunkMetaData = null;
+  }
+
+  public void endRowGroup(long memSize) throws IOException {
```
Comment (kr11) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java]: The IOException is unnecessary.

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+import java.util.List;
+import java.util.Map;
+
+/**
+ * TSFileIOWriter is used to construct metadata and write data stored in memory to output stream.
+ *
+ * @author kangrong
+ */
+public class TsFileIOWriter {
+  public static final String MAGIC_STRING = "TsFilev0.0.1";
```
Comment (kr11) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java]: The magic_string are not consistent with the actual version.

File: src/test/java/cn/edu/tsinghua/tsfile/file/metadata/TimeSeriesMetadataTest.java [hunk 18]
```diff
@@ -48,7 +48,7 @@ public void testConvertToThrift() {
   @Test
   public void testConvertToTSF() {
     for (DataType dataType : DataType.values()) {
-      TimeSeries timeSeries = new TimeSeries(measurementUID, dataType, deltaObjectType);
+      TimeSeries timeSeries = new TimeSeries(measurementUID, dataType, "");
```
Comment (kr11) [inline src/test/java/cn/edu/tsinghua/tsfile/file/metadata/TimeSeriesMetadataTest.java]: The definition of TimeSeries.thrift could be changed together in this pull request where the third parameter of TimeSeries is removed

File: src/test/java/cn/edu/tsinghua/tsfile/timeseries/demo/WriteTsfileDemo.java [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+import cn.edu.tsinghua.tsfile.file.metadata.enums.TSDataType;
+import cn.edu.tsinghua.tsfile.file.metadata.enums.TSEncoding;
+import cn.edu.tsinghua.tsfile.timeseries.write.TsFileWriter;
+import cn.edu.tsinghua.tsfile.timeseries.write.desc.MeasurementDescriptor;
+import cn.edu.tsinghua.tsfile.timeseries.write.exception.WriteProcessException;
+import cn.edu.tsinghua.tsfile.timeseries.write.record.DataPoint;
+import cn.edu.tsinghua.tsfile.timeseries.write.record.TSRecord;
+import cn.edu.tsinghua.tsfile.timeseries.write.record.datapoint.FloatDataPoint;
+
+public class WriteTsfileDemo {
```
Comment (kr11) [inline src/test/java/cn/edu/tsinghua/tsfile/timeseries/demo/WriteTsfileDemo.java]: The class name should be WriteTs`F`ileFdemo

File: src/test/java/cn/edu/tsinghua/tsfile/timeseries/demo/WriteTsfileDemo.java [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+
+	public static void main(String[] args) throws WriteProcessException, IOException {
+		TsFileWriter tsFileWriter=new TsFileWriter(new File("test.ts"));
+		tsFileWriter.addMeasurement(new MeasurementDescriptor("cpu_utility", TSDataType.FLOAT, TSEncoding.TS_2DIFF));
+		tsFileWriter.addMeasurement(new MeasurementDescriptor("memory_utility", TSDataType.FLOAT, TSEncoding.TS_2DIFF));		
+		TSRecord tsRecord=new TSRecord(1000, "hxd");
```
Comment (kr11) [inline src/test/java/cn/edu/tsinghua/tsfile/timeseries/demo/WriteTsfileDemo.java]: `hxd` seems informal

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java (hunk 17 above)
Comment (jixuan1989) [reply src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java]: I do not think this string needs to keep up with our software version. 
Actually,  it only needs to be changed when our serialization method is changed.

File: src/test/java/cn/edu/tsinghua/tsfile/file/metadata/TimeSeriesMetadataTest.java (hunk 18 above)
Comment (jixuan1989) [reply src/test/java/cn/edu/tsinghua/tsfile/file/metadata/TimeSeriesMetadataTest.java]: Will do it in another PR.

**Pull request merged**